File: HealthTrac/HealthTrac/Models/Membership.cs
﻿using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using System.ComponentModel.DataAnnotations;

namespace HealthTrac.Models
{
    public enum MembershipStatus
    {
        BANNED, MEMBER, INACTIVE, ADMIN
    }

    public class Membership : DateTrackingModel
    {
        public long ID { get; set; }
        public long TeamID { get; set; }
        public string UserID { get; set; }
        public MembershipStatus MembershipStatus { get; set; }
        public virtual Team Team { get; set; }
        public virtual User User { get; set; }
    }
}

File: HealthTrac_Mobile/app/src/main/java/com/raik383h_group_6/healthtracmobile/model/Membership.java
package com.raik383h_group_6.healthtracmobile.model;

import com.google.gson.annotations.SerializedName;

import java.util.Date;

public class Membership {
    @SerializedName("ID") private long id;

    public long getId() {
        return id;
    }

    public void setId(long id) {
        this.id = id;
    }

    public long getTeamID() {
        return teamID;
    }

    public void setTeamID(long teamID) {
        this.teamID = teamID;
    }

    public String getUserID() {
        return userID;
    }

    public void setUserID(String userID) {
        this.userID = userID;
    }

    public Date getDateCreated() {
        return dateCreated;
    }

    public void setDateCreated(Date dateCreated) {
        this.dateCreated = dateCreated;
    }

    public Date getDateModified() {
        return dateModified;
    }

    public void setDateModified(Date dateModified) {
        this.dateModified = dateModified;
    }

    public MembershipStatus getMembershipStatus() {
        return membershipStatus;
    }

    public void setMembershipStatus(MembershipStatus membershipStatus) {
        this.membershipStatus = membershipStatus;
    }

    private long teamID;
    private String userID;
    private Date dateCreated, dateModified;
    @SerializedName("MembershipStatus") private MembershipStatus membershipStatus;

    public enum MembershipStatus {
        @SerializedName("0")
        BANNED,
        @SerializedName("1")
        MEMBER,
        @SerializedName("2")
        INACTIVE,
        @SerializedName("3")
        ADMIN
    }
}
